src/sagemaker/jumpstart/accessors.py

```diff
@@ -0,0 +1,141 @@
+    """Static class for storing the JumpStart models cache."""
+
+    _cache: Optional[cache.JumpStartModelsCache] = None
+    _curr_region = JUMPSTART_DEFAULT_REGION_NAME
+
+    _cache_kwargs: Dict[str, Any] = {}
+
+    @staticmethod
+    def _validate_and_mutate_region_cache_kwargs(
+        cache_kwargs: Optional[Dict[str, Any]] = None, region: Optional[str] = None
+    ) -> Dict[str, Any]:
+        """Returns cache_kwargs with region argument removed if present.
+
+        Raises:
+            ValueError: If region in `cache_kwargs` is inconsistent with `region` argument.
+
+        Args:
+            cache_kwargs (Optional[Dict[str, Any]]): cache kwargs to validate.
+            region (str): The region to validate along with the kwargs.
+        """
+        cache_kwargs_dict = {} if cache_kwargs is None else cache_kwargs
+        assert isinstance(cache_kwargs_dict, dict)
+        if region is not None and "region" in cache_kwargs_dict:
+            if region != cache_kwargs_dict["region"]:
+                raise ValueError(
+                    f"Inconsistent region definitions: {region}, {cache_kwargs_dict['region']}"
+                )
+            del cache_kwargs_dict["region"]
+        return cache_kwargs_dict
+
+    @staticmethod
+    def get_model_header(region: str, model_id: str, version: str) -> JumpStartModelHeader:
+        """Returns model header from JumpStart models cache.
+
+        Args:
+            region (str): region for which to retrieve header.
+            model_id (str): model id to retrieve.
+            version (str): semantic version to retrieve for the model id.
+        """
+        cache_kwargs = JumpStartModelsCache._validate_and_mutate_region_cache_kwargs(
+            JumpStartModelsCache._cache_kwargs, region
+        )
+        if JumpStartModelsCache._cache is None or region != JumpStartModelsCache._curr_region:
+            JumpStartModelsCache._cache = cache.JumpStartModelsCache(region=region, **cache_kwargs)
+            JumpStartModelsCache._curr_region = region
+        assert JumpStartModelsCache._cache is not None
+        return JumpStartModelsCache._cache.get_header(model_id, version)
+
+    @staticmethod
+    def get_model_specs(region: str, model_id: str, version: str) -> JumpStartModelSpecs:
+        """Returns model specs from JumpStart models cache.
+
+        Args:
+            region (str): region for which to retrieve header.
+            model_id (str): model id to retrieve.
+            version (str): semantic version to retrieve for the model id.
+        """
+        cache_kwargs = JumpStartModelsCache._validate_and_mutate_region_cache_kwargs(
+            JumpStartModelsCache._cache_kwargs, region
+        )
+        if JumpStartModelsCache._cache is None or region != JumpStartModelsCache._curr_region:
+            JumpStartModelsCache._cache = cache.JumpStartModelsCache(region=region, **cache_kwargs)
+            JumpStartModelsCache._curr_region = region
```

Comment:
nit: @evakravi This section is same as lines 80-82 above, can we refactor this to avoid duplicacy?